File: Python/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        d = {}
        for i in range(len(nums)):
            if nums[i] in d:
                return [d[nums[i]], i]
            d[target - nums[i]] = i


# nums = [3,2,4]
# target = 6
# print(twoSum(nums, target))

File: Go/two_sum.go
package Go

func twoSum(nums []int, target int) []int {
	seen := make(map[int]int)
	for i, num := range nums {
		if j, ok := seen[num]; ok {
			return []int{j, i}
		}
		seen[target-num] = i
	}
	return nil
}
